### PR TITLE
Finalize high resolution templates support

### PR DESCRIPTION
**Pull request by zonca**
templates with high resolution were implemented in #20 and merged here.
This needs some testing, minor improvements and docs.

### CODE REVIEW

**Comment by zonca** [comment]
@NicolettaK in order to remove the 512 fits files from the repo history I had to force-push this branch, so you cannot use anymore your older `high_res_template` branch but you have to start again from this one.

**Comment by zonca** [comment]
@NicolettaK actually I created a dedicated page in the docs and filled in details about the software, you can add details about the models, see `docs/highres_templates.rst`

**Comment by NicolettaK** [comment]
Thanks @zonca !
I'll work on docs in the next days. 
Meanwhile I think you can start generating maps using these models. Do you need anything else from my side?

**Comment by zonca** [comment]
I will be able to work on sims next week, if you can get the documentation before then better.
All those models are deterministic, right? so I can run 1 "realization" each at 4096 and one at 512. For the 512 should I run with `small_scale` True or False?

**Comment by NicolettaK** [comment]
Yes they are deterministic, therefore one realization is fine. I would run with `small_scale = False` for SAT.

**Comment by zonca** [comment]
ok, I have implemented also file-listing for the folder at NERSC, so people can browse and download the models http://portal.nersc.gov/project/cmb/so_pysm_models_data/

File: docs/highres_templates.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+Details about individual models
+===============================
+
+## Dust
+
+* **SO_d0**: Thermal dust is modeled as a single-component modified black body (mbb).
+There is no spatial variation of temperature and emissivity in the sky: :math:`T=19.6` K and
+:math:`\beta_d=1.53`.
```
Comment (zonca) [inline docs/highres_templates.rst]: @NicolettaK for each of the models, (as you did for free-free) can you point to the equivalent model in PySM? If instead you changed anything, can you describe the source of your modification (i.e. Planck)?

Comment (zonca) [reply docs/highres_templates.rst]: @NicolettaK I've give you push access to the repo, so for quick edits you can directly do via web, https://github.com/simonsobs/so_pysm_models/blob/high_res_templates/docs/highres_templates.rst

File: so_pysm_models/so_models.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    return map_out
+
+
+def get_so_models(key, nside, pixel_indices=None, mpi_comm=None):
+    small_scale = key.endswith("s")
+    if small_scale:
+        nside_template = 4096
+        key = key[:-1]
+    else:
+        nside_template = 512
```
Comment (zonca) [inline so_pysm_models/so_models.py]: @NicolettaK I am proposing that we address the 4096 model as `SO_d0s` and the 512 as `SO_d0`, is this ok? would you prefer another suffix?

**Pull request merged**